sdk/src/main/java/com/mapbox/maps/MapProvider.kt

```diff
@@ -50,7 +52,9 @@ internal object MapProvider {
         paramsProvider(context, accessToken, MapboxModuleType.MapTelemetry)
       }
     }
-    mapTelemetry.onAppUserTurnstileEvent()
+    Handler(Looper.getMainLooper()).post {
```

Comment:
LGTM in general, maybe even consider lifting the event sending by 1-2 seconds?

Reply:
I think we should try to send the event as soon as possible, if user finishes the activity the turnstile event might never be sent

Reply:
I know, that several sdk send data in another process and try to make it as background service, listen when it might be effective for battery (when user charge phone) and many other things.

but for as it will over engineering

Reply:
That would be a totally new feature and we'd need to plan it out...

Reply:
Deferred telemetry is planned in common sdk, we should invoke the call as soon as possible.